src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -8332,8 +8332,8 @@ class StubGenerator: public StubCodeGenerator {
                                                 SharedRuntime::throw_delayed_StackOverflowError));
 
     // Initialize table for copy memory (arraycopy) check.
-    if (UnsafeCopyMemory::_table == nullptr) {
-      UnsafeCopyMemory::create_table(8);
+    if (UnsafeMemoryAccess::_table == nullptr) {
+      UnsafeMemoryAccess::create_table(18);
```

Comment:
Needs comment explaining 18 number

Reply:
Hmmm...  There was no comment explaining the 8 number :-).  I added 10 to the table size because I knew I was going to add 7 places where a mark was required.  I left 3 for safety.

The algorithm has since changed, so I changed this to:

`UnsafeCopyMemory::create_table(8 + 4); // 8 for copyMemory; 4 for setMemory`

I did a similar change to all other table creation numbers.

Reply:
Well, the 8 number wasn't explained either :-).  I took what was there and added 10, since the original algorithm required 7 marks for setMemory.  I added 3 for safety.  The algorithm changed so only 4 are needed, so I changed this to:

`UnsafeCopyMemory::create_table(8 + 4); // 8 for copyMemory; 4 for setMemory`

Hope this is clearer.